lib/support/CMakeLists.txt

```diff
@@ -182,6 +182,8 @@ check_cxx_source_compiles (
 set (CMAKE_THREAD_PREFER_PTHREAD TRUE)
 find_package (Threads)
 set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
+# the pthread_setname_np is defined in the musl-libc library.
+list(APPEND CMAKE_REQUIRED_LIBRARIES "c_elf")
```

Comment:
What’s this?

Reply:
Normally, the `pthread_setname_np` and `pthread_getname_np` functions are defined in the libpthread.a library. However, musl puts everything into libc.a/so to avoid memory bloat. Therefore, I added the this line to link musl-prepo libc_elf.a library when checking the `pthread_setname_np` and `pthread_getname_np` functions.

Reply:
> musl puts everything into libc.a/so

If the functions are in libc then there shouldn’t be anything to do since this is linked by default isn’t it? If we’re not linking the standard libraries by default then other feature tests could be incorrectly failing.

Reply:
>
> If the functions are in libc then there shouldn’t be anything to do since this is linked by default isn’t it?
>
Yes, the libc_elf.a is linked by default in the `llvm/utils/repo/repo.cmake`.

>
> If we’re not linking the standard libraries by default then other feature tests could be incorrectly failing due to the missing library.
>
I checked the generated `CMakeError.log` file. The library libc_elf.a is added to the link command, but it is before the tested object file ` CMakeFiles/cmTC_82a30.dir/src.cxx.o.elf`. For example:
`-lc_elf CMakeFiles/cmTC_82a30.dir/src.cxx.o.elf`. However, the order in which libraries are linked caused errors in GNU ld. It explained [Link order of libraries](https://www.linuxtopia.org/online_books/an_introduction_to_gcc/gccintro_18.html).

After added `list(APPEND CMAKE_REQUIRED_LIBRARIES "c_elf")`, the `c_elf` library is added into the end of link command line, e.g. `...-lc_elf CMakeFiles/cmTC_0ff2d.dir/src.cxx.o.elf -lc_elf`. The build and link are fine. However, we can change the link order  in the repo.cmake file to solve the issue. I have removed the line `list(APPEND CMAKE_REQUIRED_LIBRARIES "c_elf")` in the commit https://github.com/SNSystems/pstore/pull/93/commits/ed5cf97fa57af25655999699db16ec75c2c7fe96 and modified the repo.cmake in the [PR#149](https://github.com/SNSystems/llvm-project-prepo/pull/149) .